invenio_rdm_records/records/dumpers.py

```diff
@@ -57,5 +65,59 @@ def load(self, data, record_cls):
         parent_data = dict_lookup(data, self.keys, parent=True)
 
         # `None` covers the cases where exceptions were raised in _dump
-        parent_data.pop(f"{self.key}_start", None)
-        parent_data.pop(f"{self.key}_end", None)
+        parent_data.pop(self.range_key, None)
+
+
+class EDTFListDumperExt(ElasticsearchDumperExt):
+    """Elasticsearch dumper extension for support of EDTF date lists.
+
+    It iterates over the items at the specified field in the
+    document and adds dictionaries holding the values
+    :code:`{"lte": lower_strict, "gte": upper_strict}`, as parsed
+    from their EDTF fields (specified by the key).
+    These values correspond to the lower strict and upper strict bounds.
+    They are required for sorting.
+    In an ascending sort (most recent last), sort by publication_date_start
+    to get a natural sort order. In a descending sort (most recent first),
+    sort by publication_date_end.
+    """
+
+    def __init__(self, list_field, key):
+        """Constructor.
+
+        :param list_field: dot separated path to the array to process.
+        :param key: name of the EDTF field in each array item.
```

Comment:
minor: should specify it has to be the same name in each item of the list, that `key` does not accept another list.